File: Module 1/C# II/homework_2_c_sharp_due_25.11.2016/04. Appearance count/AppearanceCount.cs
/*
Appearance count
Description
Write a method that counts how many times given number appears in a given array. 
Write a test program to check if the method is working correctly.

Input
On the first line you will receive a number N - the size of the array
On the second line you will receive N numbers separated by spaces - the numbers in the array
On the third line you will receive a number X

Output
Print how many times the number X appears in the array

Constraints
1 <= N <= 1024
Time limit: 0.1s
Memory limit: 16MB

Sample tests

Input
8
28 6 21 6 -7856 73 73 -56
73

Output
2
*/

using System;

class AppearanceCount
{
    static void Main()
    {
        int n = int.Parse(Console.ReadLine());
        string[] input = Console.ReadLine().Split(' ');
        int[] array = new int[n];
        for (int i = 0; i < n; i++)
        {
            array[i] = int.Parse(input[i]);
        }

        int x = int.Parse(Console.ReadLine());

        Console.WriteLine(Count(n, array, x));
    }

    private static int Count(int n, int[] array, int x)
    {
        int result = 0;
        foreach (var item in array)
        {
            if (item == x) result++;
        }

        return result;
    }
}

File: Module 1/JS I - Fundamentals/homework_5_due_13.01.2017/problem_04/AppearanceCount.js
// Problem 04. Appearance Count - BGCoder format
function solve(args) {
	let count = +args[0],
		arr = args[1].split(' ').map(Number),
		element = +args[2];
	
	function CountAppearances(arr, val) {
		let result = 0,
			len = arr.length;
		
		for (let i = 0; i < len; i += 1) {
			if (arr[i] === element) {
				result += 1;
			}
		}
		
		return result;
	}
	
	let result = CountAppearances(arr, element);		
	return (result);
}